### PR TITLE
Add Protocols and Shasums to the provider metadata file generated

### PR DESCRIPTION
**Pull request by RLRabinowitz**
Download all manifest and shasum files in parallel, you use the information from them to fill the "protocols" and "shasum" parts of the provider metadata file

### CODE REVIEW

File: src/internal/repository-metadata-files/provider/artifacts.go
Comment (cam72cam) [inline src/internal/repository-metadata-files/provider/artifacts.go:99]: I think this is fine for now, we probably want to make sure we limit the concurrency as to not hit rate limits as we ramp up.

**Review by cam72cam — APPROVED** [review]

File: src/internal/repository-metadata-files/provider/build.go
Comment (RLRabinowitz) [reply src/internal/repository-metadata-files/provider/build.go:26]: I think that makes sense. I'll do that now

Comment (RLRabinowitz) [reply src/internal/repository-metadata-files/provider/build.go]: Done in 73a2f6652df10c00638d0a5847a42a54f0052475

**Review by cam72cam — APPROVED** [review]

File: src/internal/github/assets.go
Comment (Yantrio) [inline src/internal/github/assets.go:21]: I've raised this for after we merge this PR: https://github.com/opentofu/registry-stable/issues/15

I think we should be using slog everywhere possible

File: src/internal/github/assets.go
Comment (Yantrio) [inline src/internal/github/assets.go:25]: Let's clean this up after (//cc @cam72cam / @kislerdm ), but i dont see any reason here to return named properties, 

we could easily just return `nil, err` here.

File: src/internal/github/assets.go
Comment (Yantrio) [inline src/internal/github/assets.go:43]: I feel like this code has come across from the old registry,and I was not a fan of it back then because the caller has to be aware that they need to be closing the body. 

Maybe we could now have this return a string/[]byte instead?

File: src/internal/repository-metadata-files/provider/build.go
Comment (Yantrio) [inline src/internal/repository-metadata-files/provider/build.go:115]: For cleaning up after: we should probably have something to read/write metadata that can be shared between this package and the v1 renderer.

**Review by cube2222 — APPROVED** [review]

**Review by Yantrio — APPROVED** [review]
Mostly nitpicks, but as we're looking to move fast im happy with the logic here and I'll approve assuming we'll clean up after.

File: src/internal/github/assets.go
Comment (RLRabinowitz) [reply src/internal/github/assets.go:21]: Yes I think we'll just have a cleanup task for that

File: src/internal/github/assets.go
Comment (RLRabinowitz) [reply src/internal/github/assets.go]: Done in 8abba6412fe3fb6d26eee823a5d2e25f3b55a626

File: src/internal/github/assets.go
Comment (RLRabinowitz) [reply src/internal/github/assets.go]: Done in 8abba6412fe3fb6d26eee823a5d2e25f3b55a626

**Pull request merged**
